### PR TITLE
cleaning up some lint complaints

### PR DESCRIPTION
**Pull request by JohnRoesler**
Huge fan of this project (but I haven't looked at the codebase before)! So, I was just poking around in the code to get familiar and made some tweaks along the way. Hopefully you find it helpful.

**Description:**  fixing various linting complaints
- structs should have either pointer or value receivers but not both (some can't be adjusted due to interfaces that are being implemented that require value receivers)
- variable name collisions with imported packages
- explicitly discard errors when not checking

**Testing:** < Describe what testing was performed and which tests were added.>
- Tests are passing locally (20,641 of them)
- golanglint passing

### CODE REVIEW

**Comment by atoulme** [comment]
How did you generate those lint errors? Any chance you can offer CI changes to generate the same coverage?

File: config/configtls/configtls_test.go [hunk 1]
```diff
@@ -513,11 +513,11 @@ func TestCertificateReload(t *testing.T) {
 			// Copy certs into a temp dir so we can safely modify them
 			certFile, err := os.CreateTemp("", "cert")
 			assert.NoError(t, err)
-			defer os.Remove(certFile.Name())
+			defer func() { _ = os.Remove(certFile.Name()) }()
```
Comment (bogdandrutu) [inline config/configtls/configtls_test.go:516]:
```suggestion
			t.Cleanup(func() { assert.NoError(t, os.Remove(certFile.Name()) })
```

Comment (bogdandrutu) [reply config/configtls/configtls_test.go:516]: Same everywhere.

**Comment by JohnRoesler** [comment]
@atoulme I will do some looking into how intellij is generating the lint complaints - it would be nice there are rules in golangci-lint that can catch them

File: confmap/provider/internal/configurablehttpprovider/provider.go [hunk 2]
```diff
@@ -96,7 +96,7 @@ func (fmp *provider) Retrieve(_ context.Context, uri string, _ confmap.WatcherFu
 	if err != nil {
 		return nil, fmt.Errorf("unable to download the file via HTTP GET for uri %q: %w ", uri, err)
 	}
-	defer resp.Body.Close()
+	defer func() { _ = resp.Body.Close() }()
```
Comment (bogdandrutu) [inline confmap/provider/internal/configurablehttpprovider/provider.go:99]: Does this impact performance on the critical path for no good reasons? I would retain from doing it unless a good reason to do so.

Comment (bogdandrutu) [reply confmap/provider/internal/configurablehttpprovider/provider.go:99]: Same everywhere in no-test code.

**Review by bogdandrutu — COMMENTED** [review]
I would suggest:
1. Split from test vs non-test code;
2. For non-test let's see what impact performance what does not;

File: confmap/provider/internal/configurablehttpprovider/provider.go (hunk 2 above)
Comment (JohnRoesler) [reply confmap/provider/internal/configurablehttpprovider/provider.go:99]: I was curious, so created a benchmark to compare the two behaviors:

and in this simple example, explicitly ignoring the error in a closure is actually more performant (now i want to know why that is):

https://gist.github.com/JohnRoesler/d3bf89d60630bc60e6afd8d14ab88e90

**Pull request closed without merging**